src/runtime_src/core/common/xclbin_parser.cpp

```diff
@@ -258,22 +259,30 @@ get_kernel_id(const pt::ptree& xml_kernel, const std::string& element)
 }
 
 // Determine the address range from kernel xml entry
+// This address_range is used as cu_size in the zocl. So, we are getting it from CU instance.
 static size_t
 get_address_range(const pt::ptree& xml_kernel)
 {
   constexpr auto default_address_range = 64_kb;
   size_t address_range = default_address_range;
-  for (auto& xml_port : xml_kernel) {
-    if (xml_port.first != "port")
-      continue;
-
-    // one AXI slave port per kernel
-    if (xml_port.second.get<std::string>("<xmlattr>.mode") == "slave") {
-      address_range = convert(xml_port.second.get<std::string>("<xmlattr>.range"));
-      break;
+  try {
+    for (auto& xml_inst : xml_kernel) {
+      if (xml_inst.first != "instance")
+        continue;
+      for (auto& xml_remap : xml_inst.second) {
+        if (xml_remap.first != "addrRemap")
+          continue;
+        address_range = convert(xml_remap.second.get<std::string>("<xmlattr>.range"));
+        if (address_range != 0)
+          return address_range;
+        else
+          return default_address_range;
+      }
```

Comment:
This is not the same behavior.  The value returned is not the same after this change.   There are xclbins with kernel attribute "range=0xValue"  but same xclbins do not have range attribute.   I will vaguer that you cannot make this change.

Reply:
Thank you Soren for your inputs and for pointing that out. I haven't considered the case where xclbins might not have the range attribbute. I will make the necessary adjustments to ensure the behaviour remains consistent.

Reply:
Hi Manoj,
As discussed offline, Please remove this change and ask users to have their xclbin's updated with proper range information of slave,

Reply:
Hi Vamshi, Soren,

The Changes have been made as suggest. could you pelase review them once again? Please let me know if any further changes are required.